### PR TITLE
Update search Android bindings

### PR DESCRIPTION
**Pull request by Yenel**

### CODE REVIEW

**Review by jgandres — DISMISSED** [review]

File: bindings/java/nz/mega/sdk/MegaApiJava.java [hunk 1]
```diff
@@ -8764,99 +8875,266 @@ public long getBandwidthOverquotaDelay() {
...
+     *
+     *                     - MegaApi::ORDER_FAV_DESC = 20
+     *                     Sort nodes with favourite attr last
+     *
+     * @return List of nodes that contain the desired string in their name
      */
-    public ArrayList<MegaNode> search(String searchString, int order) {
-        return nodeListToArray(megaApi.search(searchString, order));
+    public ArrayList<MegaNode> searchOnInShares(String searchString, MegaCancelToken cancelToken, int order) {
+        return nodeListToArray(megaApi.search(searchString, cancelToken, order));
```
Comment (javiergm1983) [inline bindings/java/nz/mega/sdk/MegaApiJava.java:9129]: This binding should call to `MegaApi::searchOnInShares()`
```suggestion
        return nodeListToArray(megaApi.searchOnInShares(searchString, cancelToken, order));
```

File: bindings/java/nz/mega/sdk/MegaApiJava.java [hunk 2]
```diff
@@ -8865,74 +9143,164 @@ public long getBandwidthOverquotaDelay() {
...
-     * - MegaApi::ORDER_ALPHABETICAL_DESC = 10
-     * Sort in alphabetical order, descending
+     *                     - MegaApi::ORDER_FAV_DESC = 20
+     *                     Sort nodes with favourite attr last
      *
      * @return List of nodes that contain the desired string in their name
      */
-    public ArrayList<MegaNode> search(String searchString, MegaCancelToken cancelToken, int order) {
+    public ArrayList<MegaNode> searchOnOutShares(String searchString, MegaCancelToken cancelToken, int order) {
         return nodeListToArray(megaApi.search(searchString, cancelToken, order));
```
Comment (javiergm1983) [inline bindings/java/nz/mega/sdk/MegaApiJava.java:9213]: This binding should call to MegaApi::searchOnOutShares()
```suggestion
        return nodeListToArray(megaApi.searchOnOutShares(searchString, cancelToken, order));
```

File: bindings/java/nz/mega/sdk/MegaApiJava.java [hunk 3]
```diff
@@ -8865,74 +9143,164 @@ public long getBandwidthOverquotaDelay() {
...
+     *
+     *                     - MegaApi::ORDER_FAV_DESC = 20
+     *                     Sort nodes with favourite attr last
      *
      * @return List of nodes that contain the desired string in their name
      */
-    public ArrayList<MegaNode> search(String searchString) {
-        return nodeListToArray(megaApi.search(searchString));
+    public ArrayList<MegaNode> searchOnPublicLinks(String searchString, MegaCancelToken cancelToken, int order) {
+        return nodeListToArray(megaApi.search(searchString, cancelToken, order));
```
Comment (javiergm1983) [inline bindings/java/nz/mega/sdk/MegaApiJava.java:9297]: This binding should call to MegaApi::searchOnPublicLinks()
```suggestion
        return nodeListToArray(megaApi.searchOnPublicLinks(searchString, cancelToken, order));
```

**Review by javiergm1983 — CHANGES_REQUESTED** [review]
Hi @Yenel there are incorrect bindings, please review it.

**Comment by Yenel** [comment]
Hi @javiergm1983, the bindings you pointed out are now fixed. Please check it again. Thanks.

**Review by javiergm1983 — APPROVED** [review]
No everything seems ok.

**Comment by dwmega** [comment]
trigger compilation

**Pull request merged**